resources/generate-money-factory.php

```diff
@@ -3,59 +3,65 @@
 require __DIR__.'/../vendor/autoload.php';
 
 use Money\Currencies;
+use Money\Currency;
 
-$buffer = <<<PHP
+(static function (): void {
+    $buffer = <<<'PHP'
 <?php
 
+declare(strict_types=1);
+
 namespace Money;
 
+use InvalidArgumentException;
+
 /**
  * This is a generated file. Do not edit it manually!
  *
 PHPDOC
+ *
+ * @psalm-immutable
  */
 trait MoneyFactory
 {
     /**
      * Convenience factory method for a Money object.
      *
      * <code>
-     * \$fiveDollar = Money::USD(500);
+     * $fiveDollar = Money::USD(500);
      * </code>
      *
-     * @param string \$method
-     * @param array  \$arguments
+     * @param string $method
+     * @param array  $arguments
+     * @psalm-param empty $arguments
      *
-     * @return Money
+     * @throws InvalidArgumentException If amount is not integer(ish).
      *
-     * @throws \InvalidArgumentException If amount is not integer(ish)
+     * @psalm-pure
      */
-    public static function __callStatic(\$method, \$arguments)
+    public static function __callStatic(string $method, array $arguments): Money
     {
-        return new Money(\$arguments[0], new Currency(\$method));
+        return new Money($arguments[0], new Currency($method));
     }
 }
 
 PHP;
 
-$methodBuffer = '';
+    $methodBuffer = '';
 
-$currencies = new Currencies\AggregateCurrencies([
-    new Currencies\ISOCurrencies(),
-    new Currencies\BitcoinCurrencies(),
-]);
+    $currencies = iterator_to_array(new Currencies\AggregateCurrencies([
+        new Currencies\ISOCurrencies(),
+        new Currencies\BitcoinCurrencies(),
+    ]));
 
-$currencies = iterator_to_array($currencies);
+    usort($currencies, static fn (Currency $a, Currency $b): int => strcmp($a->getCode(), $b->getCode()));
```

Comment:
Couldn't this be done as a method Inthe iterator object ? Allow more readability and avoid altogether the need for iterator to array ?

Reply:
This was only re-aligned, not changed: will keep it as-is, since it's not a runtime component of the lib, and it has no actual tests besides running it manually.